app/api/routes/news.py

```
@@ -42,3 +42,27 @@ async def get_news(
             status_code=status.HTTP_503_SERVICE_UNAVAILABLE,
             detail=f"Error fetching news: {str(e)}",
         )
+
+
+@router.get("/{article_id}", response_model=SingleNewsArticleResponse)
+async def get_news_by_id(article_id: str):
+    article = await fetch_news_by_id(article_id)
+    return SingleNewsArticleResponse(article=article)
+    try:
+        article = await fetch_news_by_id(artilce_id)
```

Ruff (0.11.9)

Undefined name `artilce_id`

(F821)